src/libraries/Common/src/System/Net/Internals/SocketExceptionFactory.cs

```diff
@@ -7,29 +7,9 @@ namespace System.Net.Internals
 {
     internal static partial class SocketExceptionFactory
     {
-        private sealed class ExtendedSocketException : SocketException
-        {
-            private readonly EndPoint? _endPoint;
-
-            public ExtendedSocketException(int errorCode, EndPoint? endPoint)
-                : base(errorCode)
-            {
-                _endPoint = endPoint;
-            }
-
-            public ExtendedSocketException(SocketError socketError, int platformError)
-                : base((int)socketError)
-            {
-                HResult = platformError;
-            }
-
-            public override string Message =>
-                (_endPoint == null) ? base.Message : base.Message + " " + _endPoint.ToString();
-        }
-
         public static SocketException CreateSocketException(int socketError, EndPoint? endPoint)
         {
-            return new ExtendedSocketException(socketError, endPoint);
+            return new SocketException(socketError, endPoint?.ToString());
```

Comment:
`Message` used to contain more info than `endPoint.ToString()`. I think we should utilize `SocketException` / `SocketErrorPal` implementation details.

Reply:
Alright, so creating a custom message with the endpoint info in this method should be enough, then?

Reply:
Maybe creating a temporary object for SocketException with `SocketException(int)` overload and we can transfer the message from there? Instead of moving the same implementation here.

Reply:
Depends on the contents of the custom message. We depend on the OS to query the error string from the native error code:

https://github.com/dotnet/runtime/blob/d196d850079ba88e904123c725e0f358142349c7/src/libraries/System.Net.Primitives/src/System/Net/SocketException.cs#L34

https://github.com/dotnet/runtime/blob/d196d850079ba88e904123c725e0f358142349c7/src/libraries/System.Private.CoreLib/src/System/ComponentModel/Win32Exception.cs#L31